Sources/Purchasing/StoreKit2/Observer Mode/StoreKit2ObserverModePurchaseDetector.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2ObserverModePurchaseDetector.swift
+//
+//  Created by Will Taylor on 5/1/24.
+
+import Foundation
+import StoreKit
+
+/// A delegate protocol for handling verified transactions in observer mode.
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
```

Comment:
With Swift Macros we'll eventually be able to re-replace these with stuff like `"STOREKIT_2_COMPATIBLE_OS"` right? I know they're not as powerful as C pre-processor macros so I've been wondering

Reply:
we used to have that back when the SDK was ObjC and I definitely miss it

Reply:
That we be amazing - it'd be way too easy to accidentally copy/paste the wrong one somewhere. Hopefully we can make that work with Macros!